src/main/java/refactoringml/RefactoringAnalyzer.java

```diff
@@ -220,47 +298,44 @@ private RefactoringCommit calculateCkMetrics(ImmutablePair<String,String> refact
 
 			// finally, if it's a field refactoring, we then only have class + field
 			FieldMetric fieldMetrics = null;
-			if(isAttributeLevelRefactoring(refactoring)) {
+			if (isAttributeLevelRefactoring(refactoring)) {
 				String refactoredField = getRefactoredVariableOrAttribute(refactoring);
 
 				int totalAppearances = ck.getMethods().stream()
-						.map(x -> x.getFieldUsage().get(refactoredField) == null ? 0 : x.getFieldUsage().get(refactoredField))
+						.map(x -> x.getFieldUsage().get(refactoredField) == null ? 0
+								: x.getFieldUsage().get(refactoredField))
 						.mapToInt(Integer::intValue).sum();
 
 				fieldMetrics = new FieldMetric(refactoredField, totalAppearances);
 			}
 
 			// assemble the final object
-			RefactoringCommit refactoringCommit = new RefactoringCommit(
-					project,
-					commitMetaData,
-					enforceUnixPaths(ck.getFile()).replace(tempDir, ""),
-					cleanedCkClassName,
-					refactoring.getRefactoringType().getDisplayName(),
-					refactoringTypeInNumber(refactoring),
-					refactoringSummary,
-					classMetric,
-					methodMetrics,
-					variableMetrics,
-					fieldMetrics);
+			RefactoringCommit refactoringCommit = new RefactoringCommit(project, commitMetaData,
+					enforceUnixPaths(ck.getFile()).replace(tempDir, ""), cleanedCkClassName,
+					refactoring.getRefactoringType().getDisplayName(), refactoringTypeInNumber(refactoring),
+					refactoringSummary, classMetric, methodMetrics, variableMetrics, fieldMetrics);
 			refactorings.add(refactoringCommit);
 		});
 
 		/**
 		 * It is possible that we did not find the class among the results of CK.
 		 * Possible causes are:
 		 *
-		 *   1) Anonymous classes. It can be really hard to match anonymous classes from CK with the ones from RMiner.
-		 *      See this issue in CK: https://github.com/mauricioaniche/ck/issues/54.
+		 * 1) Anonymous classes. It can be really hard to match anonymous classes from
+		 * CK with the ones from RMiner. See this issue in CK:
+		 * https://github.com/mauricioaniche/ck/issues/54.
 		 */
-		if(refactorings.isEmpty()) {
-			log.error("CK did not find class " + refactoredClasses.getLeft() + "/" + refactoredClasses.getRight() + createRefactoringErrorState(commitMetaData.getCommitId(), project, refactoringSummary));
+		if (refactorings.isEmpty()) {
+			String refactoringErrorState = createRefactoringErrorState(commitMetaData.getCommitId(), project,
+					refactoringSummary);
+			log.error("CK did not find class {}/{} {}", refactoredClasses.getLeft(), refactoredClasses.getRight(),
+					refactoringErrorState);
 		} else {
 			for (RefactoringCommit refactoringCommit : refactorings) {
-				db.persist(refactoringCommit);
+				db.save(refactoringCommit);
```

Comment:
This applies also to the other uses of save instead of persist.

Why would you use a save here? In my opinion, persist is the saver option:
1. it guarantees that it is not inserted outside a transaction, thus we can reroll the last transaction if necessary. Or in cases of crashes never commit the transaction and thereby, keep the DB cleaner.
2. The id created for the object is not created when it is called, but when it is flushed. Thus, we either have no race conditions between multiple workers trying to assign the same id to an object or the database needs less locks.

Reply:
I thought save was just a hibernate specific way of persist and since we are using hibernate specific sessions and sessionfactories rather than entitymanagers and entitymanagerfactories I figured it was a good idea to be consistent and use hibernate specific methods as well. After looking it up it seems that it does not just have extra functionality of returning an id flushing and detaching the object. I will change this back to persist. Maybe it is an idea to generify the session to entitymanager though since we are mostly only using entitymanager methods anyway.